File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/Conditional.java
// Conditional.java

public class Conditional extends Ternary
{
	public Conditional(Expression aOperand1, Expression aOperand2, Expression aOperand3)
	{
		super(aOperand1, aOperand2, aOperand3);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/lab/conditional.cpp
// conditional.cpp

#include "conditional.h"
class expression;
#include "expression_visitor.h"
#include "ternary.h"

conditional::conditional(expression * a_operand1, expression * a_operand2, expression * a_operand3)
: ternary(a_operand1, a_operand2, a_operand3)
{
}

void conditional::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}
